Userland/Libraries/LibGUI/Action.cpp

```diff
@@ -225,7 +225,12 @@ void Action::set_group(Badge<ActionGroup>, ActionGroup* group)
 
 void Action::set_icon(Gfx::Bitmap const* icon)
 {
+    if (m_icon == icon)
+        return;
     m_icon = icon;
+    for_each_toolbar_button([&](auto& button) {
+        button.set_icon(icon);
+    });
```

Comment:
Also need to do this for the menu items, similar to `set_checked`:

```
    for_each_toolbar_button([checked](auto& button) {
        button.set_checked(checked);
    });
    for_each_menu_item([checked](MenuItem& item) {
        item.set_checked(checked);
    });
```

Reply:
The `MenuItem`'s bitmap gets sent over IPC to WindowServer.

Do you think serializing that every time `MenuItem::update_window_server()` gets called makes sense, should I break out bitmap sharing into a separate IPC call or perhaps even have a separate call for every property that can change?